tasks/agent6-linux.yml

```diff
@@ -58,14 +58,29 @@
     group: "{{ datadog_group }}"
   notify: restart datadog-agent
 
+- name: Create system-probe configuration file
+  template:
+    src: system-probe.yaml.j2
+    dest: /etc/datadog-agent/system-probe.yaml
+    mode: 0644
+    owner: "root"
+    group: "root"
+
 - name: Ensure datadog-agent is running
   service:
     name: datadog-agent
     state: started
     enabled: yes
   when: not datadog_skip_running_check and datadog_enabled and not ansible_check_mode
 
-- name: Ensure datadog-agent, datadog-agent-process and datadog-agent-trace are not running
+- name: Ensure datadog-agent-sysprobe is running if enabled
+  service:
+    name: datadog-agent-sysprobe
+    state: started
+    enabled: yes
+  when: not datadog_skip_running_check and datadog_enabled and not ansible_check_mode and system_probe_config is defined and system_probe_config["enabled"]
+
+- name: Ensure datadog-agent, datadog-agent-process, datadog-agent-trace and datadog-agent-sysprobe are not running
```

Comment:
Will this still work on older agent versions that don't have the system probe bundled?

Reply:
Indeed, I don't think it will

Reply:
Good catch.  I've updated this to use the [service_facts](https://docs.ansible.com/ansible/latest/modules/service_facts_module.html#service-facts-module) module to determine if the service is present.